schism/page_info.c

```diff
@@ -255,15 +255,18 @@ static void info_draw_samples(int base, int height, int active, int first_channe
 
 	for (pos = base + 1; pos < base + height - 1; pos++, c++) {
 		song_voice_t *voice = current_song->voices + c - 1;
-
 		/* always draw the channel number */
-		if (c == selected_channel)
+		memcpy(buf, numtostr(2, c, buf), 8*sizeof(char));
```

Comment:
`numtostr` modifies `buf` inplace and returns the same pointer. This code is *undefined behavior*:

> If copying takes place between objects that overlap, the behavior is undefined.
>
> https://pubs.opengroup.org/onlinepubs/007904975/functions/memcpy.html